zinc/src/main/scala/sbt/internal/inc/MixedAnalyzingCompiler.scala

```diff
@@ -181,9 +191,16 @@ object MixedAnalyzingCompiler {
       incrementalCompilerOptions: IncOptions,
       extra: List[(String, String)]
   ): CompileConfiguration = {
-    val classpathHash = classpath map { x =>
-      FileHash.of(x, Stamper.forHash(x).hashCode)
+    val parallelClasspathHashing = classpath.toParArray.map { file =>
+      val attrs = Files.readAttributes(file.toPath, classOf[BasicFileAttributes])
+      val currentFileTime = attrs.lastModifiedTime()
```

Comment:
using size as well as modified is effectively free at his point (it's in attrs) so we could use it to be safe.

Reply:
Sure.